beta-src/src/state/game/game-api-slice.ts

```diff
@@ -375,14 +416,17 @@ const gameApiSlice = createSlice({
           unitID,
         },
         ownUnits,
+        overview: { phase },
       } = current(state);
       if (contextVars?.context?.orderStatus) {
         const orderStates = getOrderStates(contextVars?.context?.orderStatus);
         if (orderStates.Ready) {
           return;
         }
       }
-      if (inProgress) {
+      if (phase === "Builds") {
+        destroyUnit(state, clickData.payload.unitID);
+      } else if (inProgress) {
```

Comment:
You should be able to leverage your `destroyUnits` function you created and place that into `drawOrders`, this way whenever orders are updated you only have to worry about setting the correct icon states in your function.